src/sys/socket/mod.rs

```diff
@@ -885,6 +888,33 @@ pub struct Timestamps {
     pub hw_raw: TimeSpec,
 }
 
+/// These constants correspond to TLS 1.2 message types, as defined in
+/// RFC 5246, Appendix A.1
+#[cfg(any(target_os = "android", target_os = "linux"))]
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+#[repr(u8)]
+#[non_exhaustive]
+pub enum TlsGetRecordType {
```

Comment:
FWIW TLS 1.3 has a lot more of these. In particular you're missing SessionTicket(4), which is an almost-guaranteed message you'll receive when doing a TLS 1.3 connection as a post-handshake message.

[This](https://github.com/arsing/ktls-test/blob/10316091b43eece021238c9602ee312d5a886d1d/src/tls.rs#L42-L78) is the full list.

That said, and I know @asomers asked for it, but I don't think this belongs in the nix crate because it's about TLS, not libc or Linux.

Reply:
fwiw, re: your original comment:

> That said, and I know @asomers asked for it, but I don't think this belongs in the nix crate because it's about TLS, not libc or Linux.

I agree with this, but I'm willing to do whatever to get this PR merged at this point. I agree the API boundaries are kinda unclear right now, after all libc itself is a grab bag 🤷